internal/kubernetes/operator/project/project.go

```diff
@@ -134,11 +134,11 @@ func BuildAtlasProject(projectStore store.AtlasOperatorProjectStore, validator f
 		projectResult.Spec.MaintenanceWindow = maintenanceWindows
 	}
 
-	secretRef := &common.ResourceRefNamespaced{}
 	if includeSecret {
+		secretRef := &common.ResourceRefNamespaced{}
```

Comment:
Same thing here about the connection secret. It should always be there, even without any credentials